app/src/main/java/io/homeassistant/companion/android/sensors/ChargingBroadcastReceiver.kt

```diff
@@ -0,0 +1,38 @@
+package io.homeassistant.companion.android.sensors
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import java.lang.Exception
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+class ChargingBroadcastReceiver(
+    private val integrationUseCase: IntegrationUseCase
+) : BroadcastReceiver() {
+    private val ioScope = CoroutineScope(Dispatchers.IO)
+    private var updateJob: Job? = null
+    override fun onReceive(context: Context, intent: Intent?) {
+        updateJob?.cancel()
+        updateJob = ioScope.launch {
+            val batterySensorManager = BatterySensorManager().registerSensors(context)
+            integrationUseCase.updateSensors(
+                batterySensorManager.getSensors(context).toTypedArray()
+            )
+        }
+    }
```

Comment:
Rather than duplicating the logic why not call `SensorWorker.start(context)`?  That way we get all sensors updating.

Reply:
Interesting thought. I'll give that a try. It should work as long as it replaces the existing SensorWorker.

Reply:
I could even add a delay to the restart of SensorWorker which might get around is_charging somewhat. It's using a delay as a solution which generally is not the best idea imo. It probably makes more sense to update what the OS immediately knows which is that the charger_type vs trying to hack around getting is_charging to work right.

In any case I'll mess around with this soon.

Reply:
Let's try not to add more features to this PR. Instead, keep it for a future PR. It's easier to review small PRs.

Reply:
Also, if you're interested, you can join us on #devs_mobile_apps on the Home Assistant Discord server. It's where we hangout :) https://www.home-assistant.io/join-chat/

Reply:
Did you try calling `SensorWorker.start(context)` instead?  That should be included in this PR because we don't want to duplicate this logic.

Reply:
Yeah sorry see my comment I posted at the bottom of the PR. That wasn't working because it still entrusts Android to actually run the SensorWorker code. Therefore it wouldn't reliably immediately execute and could be delayed by up to 15 minutes which is the problem I had hoped to solve with this PR.

Reply:
Totally missed that.  Can we do `SensorWorker(context, WorkerParameters()).doWork()` instead?  If we can't do that then we should extract that code somewhere that we can call it independently so we don't duplicate it.

Reply:
I fully agree with the sentiment. I originally didn't love that I was duplicating logic but didn't want to start ripping things apart for a first PR. Essentially I played it safe. However I will definitely look into a way of doing it. The best way, to me, feels like taking what SensorWorker and pulling that into a class that can be called elsewhere and composed into other sources of sensor updates besides a work manager. I'll go ahead and try something out.

Reply:
Ok I added a new interface/impl `SensorUpdater` which contains that logic in one place. It is not inserted into the dagger graph. In order to do that I would need to make some significant changes that I felt were starting to get out of scope and I didn't want to cause additional delays for this functionality. I can definitely tackle this in a follow up PR where we can focus more exclusively on that. Let me know what you think.